transcom/mymove
# 21680 counseling office timeout changes

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1031521&RoomContext=TeamRoom%3A848240&concept=TeamRoom)

## [INT PR](https://github.com/transcom/mymove/pull/14242)
## [INT PR 2](https://github.com/transcom/mymove/pull/14294/files)

## Summary

When a customer creates a new move, the Counseling Office field does not immediately load. It takes a few seconds before the data is available to select.  At some times, it get a timeout error and never loads.  This is preventing any new moves to be created by the customer on Integration.
Also, there was an issue when you change Current Duty Location, it wasn't resetting the Counseling Office dropdown value, and then allowing the customer to submit with a blank Counseling office.  This has been fixed as well.

### How to test

1. Login as a Milmove customer and create a new move.
2. After picking a New Duty Location from the dropdown, go to choose a Counseling office.
3. Verify that it loads relatively quick, 2-3 seconds roughly.
4. Go back and change the New Duty Location selection.
5. Before picking a new Counseling Office, verify that Counseling Office is showing the red Required label and that Next is not enabled.
6. Add a Counseling Office selection and save.
